e2e/tests/lib/test_server_and_client_preparing/suite.go

```diff
@@ -0,0 +1,455 @@
+		return nil
+	}
+	Expect(err).ToNot(HaveOccurred(), "error response reading at container")
+	return fmt.Errorf("unexpected output creating directory: %s", text)
+}
+
+func (s *Suite) ExecuteCommandAtContainer(container *types.Container, cmd []string, extraInputToSTDIN []string) []string {
+	ctx := context.Background()
+	config := types.ExecConfig{
+		AttachStdin:  true,
+		AttachStderr: true,
+		AttachStdout: true,
+		Cmd:          cmd,
+	}
+
+	IDResp, err := s.dockerCli.ContainerExecCreate(ctx, container.ID, config)
+	Expect(err).ToNot(HaveOccurred(), "error execution at container")
+
+	resp, err := s.dockerCli.ContainerExecAttach(ctx, IDResp.ID, types.ExecStartCheck{})
+	defer resp.Close()
+	Expect(err).ToNot(HaveOccurred(), "error attaching execution at container")
+
+	go func() {
+		for _, input := range extraInputToSTDIN {
+			time.Sleep(time.Millisecond * 500)
+			resp.Conn.Write([]byte(input + "\n"))
+		}
+	}()
+
+	output := []string{}
+	var text string
+	for err == nil {
+		text, err = resp.Reader.ReadString('\n')
+		if text != "" {
+			output = append(output, text)
+		}
+	}
+
+	if err != nil && err.Error() == "EOF" {
+		fmt.Printf("command: \n %s \n ==> has been succeseed at  at container  %s \n", cmd, container.Names)
+		return output
+	}
+	Expect(err).ToNot(HaveOccurred(), "error response reading at container")
+	return nil
+}
+
+func (s *Suite) killProcessAtContainer(container *types.Container, processPid int) {
+	ctx := context.Background()
+	config := types.ExecConfig{
+		AttachStdin:  true,
+		AttachStderr: true,
+		AttachStdout: true,
+		Cmd:          []string{"/bin/bash", "-c", "kill -9 " + strconv.Itoa(processPid)},
+	}
+
+	IDResp, err := s.dockerCli.ContainerExecCreate(ctx, container.ID, config)
+	Expect(err).ToNot(HaveOccurred(), "error execution at container")
+
+	resp, err := s.dockerCli.ContainerExecAttach(ctx, IDResp.ID, types.ExecStartCheck{})
+	defer resp.Close()
+	Expect(err).ToNot(HaveOccurred(), "error attaching execution at container")
+	fmt.Println(err)
+}
+
+func (s *Suite) killAllInstancesOfProcessAtContainer(container *types.Container, processPath string) {
+	for {
+		pid := s.firstProcessPIDAtContainer(container, processPath)
+		if pid == 0 {
+			break
+		}
+		s.killProcessAtContainer(container, pid)
+	}
+}
+
+func (s *Suite) firstProcessPIDAtContainer(container *types.Container, processPath string) int {
+	ctx := context.Background()
+	config := types.ExecConfig{
+		AttachStdin:  true,
+		AttachStderr: true,
+		AttachStdout: true,
+		Cmd:          []string{"/bin/bash", "-c", "ps ax"},
+	}
+
+	IDResp, err := s.dockerCli.ContainerExecCreate(ctx, container.ID, config)
+	Expect(err).ToNot(HaveOccurred(), "error execution at container")
+
+	resp, err := s.dockerCli.ContainerExecAttach(ctx, IDResp.ID, types.ExecStartCheck{})
+	defer resp.Close()
+	Expect(err).ToNot(HaveOccurred(), "error attaching execution at container")
```

Comment:
I have a habit of deferring response closing only after the error is checked, since otherwise response can be nil and it is a common practise. I believe, it doesn't matter in this case with gomega.

Reply:
fixed